processor/src/system/mod.rs

```diff
@@ -296,3 +297,52 @@ impl System {
         }
     }
 }
+
+// EXECUTION CONTEXT
+// ================================================================================================
+
+/// Represents the ID of a memory context
```

Comment:
minor nit: let's change "memory context" to "execution context" here as well.

Reply:
done